azurerm/internal/services/network/network_connection_monitor_resource.go

```diff
@@ -200,10 +200,30 @@ func resourceNetworkConnectionMonitor() *schema.Resource {
 							},
 						},
 
+						"resource_id": {
+							Type:     schema.TypeString,
+							Optional: true,
+							Computed: true,
+							ValidateFunc: validation.Any(
+								computeValidate.VirtualMachineID,
+								logAnalyticsValidate.LogAnalyticsWorkspaceID,
+							),
+						},
+
+						"type": {
+							Type:     schema.TypeString,
+							Optional: true,
+							ValidateFunc: validation.StringInSlice([]string{
+								string(network.MMAWorkspaceMachine),
```

Comment:
Can we support the other types too? (AzureSubnet, AzureVM, AzureVNet, ExternalAddress, MMAWorkspaceNetwork)

What was the effective default here in 2020-05-01? We need to ensure this is backward compatible.

Reply:
This property doesn't exist in 2020-05-01. So I assume there is no backward compatible problem.

Reply:
Does that mean the API returns a blank value in the response when getting a resource where it wasn't set?

Reply:
Yes